lib/matplotlib/tests/test_axes.py

```diff
@@ -9166,6 +9169,46 @@ def test_axes_clear_behavior(fig_ref, fig_test, which):
     ax_test.grid(True)
 
 
+def test_axes_clear_reference_cycle():
+    def is_in_reference_cycle(start):
+        # Breadth first search. Return True if we encounter the starting node
+        to_visit = deque([start])
+        explored = set()
+        while len(to_visit) > 0:
+            parent = to_visit.popleft()
+            for child in gc.get_referents(parent):
+                if id(child) in explored:
+                    continue
+                if child is start:
+                    return True
+                explored.add(id(child))
+                to_visit.append(child)
+        return False
+    fig = Figure()
+    ax = fig.add_subplot()
+    ax.plot(np.random.rand(1000))
+    ax_children = ax.get_children()
+    fig.clear()  # This should break the reference cycle
+
+    # Care most about the objects that scale with number of points
+    line_artists = list(
+        filter(
+            lambda a: isinstance(a, Line2D) or isinstance(a, PathCollection),
+            ax_children,
+        )
+    )
```

Comment:
I think `filter` is mostly out of favour these days, plus this is shorter.
```suggestion
    line_artists = [
        a for a in ax_children
        if isinstance(a, (Line2D, PathCollection))
    ]
```

Reply:
But also, you don't add any `PathCollection`, so you could just get the line artists (there's only one) as the return value from the `ax.plot` call.

Reply:
OK, weird, there was no `scatter` there even though I opened this long after you pushed that change,